### DISCUSSION TITLE
[Help]: Having trouble getting loop to work nicely

### DISCUSSION THREAD

**Discussion by jakeRPowell** [davidjerleke/embla-carousel#546] [Help] [answered]
Hi, 

I'm after a bit of a nudge, I've got a carousel that mostly works, however at the end of the slide content there's a blank space where I'd expect perhaps duplicated slide content to be. 

https://github.com/davidjerleke/embla-carousel/assets/43782301/ed40c69a-7714-481a-8124-ee7b562446aa

I've tried a couple of different ways to add my own duplicated content in, eg a cloned version of the slides array that i absolutely positioned on the end, but this caused some weird jumping behaviour at the end. 
Another option I considered was to remove a slide from the start, and append it to the end, but again this caused weird jumping.

Can anyone point me in the right direction in terms of how you'd handle this? 

Apologies if this is the wrong forum, can provide further info if helpful, and thanks in advance!

for reference here's my carousel

```
	const [emblaRef, emblaApi] = useEmblaCarousel({
		align: "start",
		loop: true,
		slidesToScroll: 1
	});

<div className="carousel" ref={emblaRef}>
    <div className="carousel__track" ref={carouselRef}>
        {reviewsData?.reviewsData?.reviews.map((review) => (
		<CarouselSlide
			isScrolling={isScrolling}
			scrollDirection={scrollDirection}
			key={uuid()}
		>
			<Body>
				<span>{review.service.review}</span>
			</Body>
			<Name>{review.customer?.display_name}</Name>
		</CarouselSlide>
	))}
    </div>
</div>

```

and my styles 

```
export const CarouselContainer = styled.div`
	${({ theme }) => `
		.carousel {
			padding: 0 20px;
		}
		.carousel__track {
			display: flex;
		}
                .carousel__slide {
                         flex: 0 0 274px;
	            	min-width: 0;
		        margin: 0 10px;
                }
	`}
`;
```

**Comment by davidjerleke** [comment]
Hi @jakeRPowell,

Thank you for your question. As far as I can see, the library loop feature is working fine? Have you tried the [loop example](https://www.embla-carousel.com/examples/predefined/#loop) in the docs? Alternatively, you could customize your setup using the [carousel generator](https://www.embla-carousel.com/examples/generator/).

I can’t help out with fragments of code because that’s basically debugging blindfolded which is incredibly time consuming. It could literally be anything in your setup causing this. Please setup a CodeSandbox if you want me to take a look at you setup. You may fork any Sandbox in the docs or use the carousel generator.

Best,
David

**Reply by jakeRPowell** [reply, marked as answer]
Hey @davidjerleke , thanks for the quick response, I don't yet understand the how or why but I've cannibalised your example and it's working as intended now.

Cheers, 
Jake
